### ISSUE TITLE
test failed: JIT\Directed\coverage\importer\Desktop\volatilstind_il_r\volatilstind_il_r.cmd

### ISSUE THREAD

**Issue by VincentBu** [dotnet/coreclr#27326]
**Job:**
[coreclr-gcstress-extra #20191019.1 Run Test Pri1 Windows_NT x86 checked](https://dev.azure.com/dnceng/public/_build/results?buildId=394681)

**Detail:**
https://helix.dot.net/api/2019-06-17/jobs/12c2b282-2ae4-4d4c-b9bf-7fd5ce3589a5/workitems/JIT.Directed/console

**OS & Arch:**
windows x86

**Mode:**
COMPlus_TieredCompilation=0
COMPlus_GCStress=0xC
COMPlus_JitStress=2
COMPlus_ReadyToRun=0
COMPlus_ZapDisable=1

**Note:**
This failure **repros** on our windows x64 machine.

**Log:**
```
JIT\Directed\coverage\importer\Desktop\volatilstind_il_r\volatilstind_il_r.cmd [FAIL]
      
      Assert failure(PID 3472 [0x00000d90], Thread: 6740 [0x1a54]): curOffsRegs <= curOffsArgs
      
      CORECLR! scanArgRegTableI + 0x119 (0x71e875e9)
      CORECLR! EECodeManager::EnumGcRefs + 0x340 (0x71e81240)
      CORECLR! GcStackCrawlCallBack + 0x150 (0x7204d5a0)
      CORECLR! Thread::MakeStackwalkerCallback + 0x3A (0x71caf541)
      CORECLR! Thread::StackWalkFramesEx + 0x16E (0x71cb0b6a)
      CORECLR! Thread::StackWalkFrames + 0x159 (0x71cb097c)
      CORECLR! ScanStackRoots + 0x18C (0x7215bc24)
      CORECLR! GCToEEInterface::GcScanRoots + 0x104 (0x7215afa3)
      CORECLR! WKS::gc_heap::mark_phase + 0x1B1 (0x72112386)
      CORECLR! WKS::gc_heap::gc1 + 0x18A (0x7210c9ba)
          File: f:\workspace.7\_work\1\s\src\vm\eetwain.cpp Line: 2445
          Image: C:\dotnetbuild\work\12c2b282-2ae4-4d4c-b9bf-7fd5ce3589a5\Payload\CoreRun.exe
      
      
      Return code:      1
      Raw output file:      C:\dotnetbuild\work\12c2b282-2ae4-4d4c-b9bf-7fd5ce3589a5\Work\ffeac124-90e6-4d7c-b0ab-b0d533cd8dcc\Exec\JIT\Directed\Reports\JIT.Directed\coverage\importer\Desktop\volatilstind_il_r\volatilstind_il_r.output.txt
      Raw output:
      BEGIN EXECUTION
       "C:\dotnetbuild\work\12c2b282-2ae4-4d4c-b9bf-7fd5ce3589a5\Payload\corerun.exe" volatilstind_il_r.dll 
      Expected: 100
      Actual: -1073740286
      END EXECUTION - FAILED
      FAILED
      Test Harness Exitcode is : 1
```

**Comment by BruceForstall** [comment]
@briansull Can you look into this?

**Comment by briansull** [comment]
Ran this test 100 times on my Dev box, without hit this issue:

"D:\fxkit\coreclr\bin\tests\Windows_NT.x86.Checked\Tests\Core_Root\corerun.exe" volatilldind_il_r.dll

COMPlus_GCStress=0xC
COMPlus_JitStress=2
COMPlus_ReadyToRun=0
COMPlus_TieredCompilation=0
COMPlus_ZapDisable=1

**Comment by briansull** [comment]
I have a fix for this

**Comment by briansull** [comment]
Fixed with #27561
